dask_bigquery/tests/test_core.py

```diff
@@ -0,0 +1,100 @@
+import random
+
+import pandas as pd
+import pytest
+from distributed.utils_test import cluster_fixture  # noqa: F401
+from distributed.utils_test import client, loop  # noqa: F401
+from google.cloud import bigquery
+
+from dask_bigquery import read_gbq
+
+# These tests are run locally and assume the user is already athenticated.
+# It also assumes that the user has created a project called dask-bigquery.
+
+
+def gen_data(size=10):
+    records = [
+        {
+            "name": random.choice(["fred", "wilma", "barney", "betty"]),
+            "number": random.randint(0, 100),
+            "idx": i,
+        }
+        for i in range(size)
+    ]
+    return pd.DataFrame(records)
+
+
+def push_data():
+    "Push data to BigQuery using pandas gbq"
+    df = gen_data()
+
+    pd.DataFrame.to_gbq(
+        df,
+        destination_table="dataset_test.table_test",
+        project_id="dask-bigquery",
+        chunksize=5,
+        if_exists="append",
+    )
+
+    return df
+
+
+# test simple read
+def test_read_gbq(client):
+    """Test simple read of data pushed to BigQuery using pandas-gbq"""
+    try:
+        # delete data set if exists
+        bq_client = bigquery.Client()
+        bq_client.delete_dataset(
+            dataset="dask-bigquery.dataset_test",
+            delete_contents=True,
+        )
+        bq_client.close()
+    except:  # if data doesn't exisit continue is that Value Error?
+        pass
+    # create data
+    df = push_data()
+
+    ddf = read_gbq(
+        project_id="dask-bigquery", dataset_id="dataset_test", table_id="table_test"
+    )
+
+    assert ddf.columns.tolist() == ["name", "number", "idx"]
+    assert len(ddf) == 10
+    assert ddf.npartitions == 2
```

Comment:
Can we verify that the data is actually the same as the data created in `push_data` instead?  Pushing the fixture idea a little further

```python
@pytest.fixture
def df():
    ...

@pytest.fixture
def dataset(df):
    ...

def test_read_gbq(client, dataset, df):
    ddf = read_gbq(...)

    assert_eq(ddf, df)
```

Maybe there are sorting things that get in the way (is GBQ ordered?)  If so then, as you did before

```python
assert_eq(ddf.set_index("idx"), df.set_index("idx"))
```

In general we want to use assert_eq if possible.  It runs lots of cleanliness checks on the Dask collection, graph, metadata, and so on.

Reply:
Yes, I get some order issues when reading back from gbq, mainly because when I read back the default index goes from 0 to chunksize-1, where chunksize was chosen when I pushed the pandas dataframe. This was part of the reason I had as an extra column "idx".

But thanks for pointing out the `assert_eq` I forgot we had that. Although, in order for that line to work I had to do a compute on the dask dataframe. I'm assuming this is because I'm comparing a dask data frame with a pandas one.?
```python
assert_eq(ddf.set_index("idx"), df.set_index("idx"))
```

Reply:
`assert_eq` should handle comparing Dask and pandas DataFrames. What error do you get without the `compute()`?

Reply:
I get this:
```python
____________________________________________________________________ test_read_gbq _____________________________________________________________________

df =      name  number  idx
0   betty      71    0
1    fred      36    1
2   wilma      75    2
3   betty      13    3
4  ...   4
5    fred      74    5
6   wilma      69    6
7    fred      31    7
8  barney      31    8
9   betty      97    9
dataset = 'dask-bigquery.dataset_test.table_test', client = <Client: 'tcp://127.0.0.1:55212' processes=2 threads=2, memory=32.00 GiB>

    def test_read_gbq(df, dataset, client):
        """Test simple read of data pushed to BigQuery using pandas-gbq"""
        project_id, dataset_id, table_id = dataset.split(".")
    
        ddf = read_gbq(
            project_id=project_id, dataset_id=dataset_id, table_id=table_id
        )
    
        assert ddf.columns.tolist() == ["name", "number", "idx"]
        assert len(ddf) == 10
        assert ddf.npartitions == 2
        #breakpoint()
>       assert assert_eq(ddf.set_index("idx"), df.set_index("idx"))

test_core.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../../mambaforge/envs/test_gbq/lib/python3.8/site-packages/dask/dataframe/utils.py:541: in assert_eq
    assert_sane_keynames(a)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ddf = Dask DataFrame Structure:
                 name number
npartitions=2               
0              object  int64
4                 ...    ...
9                 ...    ...
Dask Name: sort_index, 22 tasks

    def assert_sane_keynames(ddf):
        if not hasattr(ddf, "dask"):
            return
        for k in ddf.dask.keys():
            while isinstance(k, tuple):
                k = k[0]
            assert isinstance(k, (str, bytes))
            assert len(k) < 100
            assert " " not in k
>           assert k.split("-")[0].isidentifier()
E           AssertionError

../../../../../mambaforge/envs/test_gbq/lib/python3.8/site-packages/dask/dataframe/utils.py:621: AssertionError

```

Reply:
Oh, interesting! Do you have any additional information in the traceback? I'm wondering what `k` is? You could try adding `--pdb` to the end of your `pytest` command, which will drop you into a `pdb` session at the point the test raises an error. You can then do `pp k` to see what `k` is

Reply:
Also xref https://github.com/dask/dask/pull/8061

Reply:
> Oh, interesting! Do you have any additional information in the traceback? I'm wondering what `k` is? You could try adding `--pdb` to the end of your `pytest` command, which will drop you into a `pdb` session at the point the test raises an error. You can then do `pp k` to see what `k` is

There is no additional in the traceback but `pp k` returns
`'dataset_test.table_test--46e9ff0148164adf1b543e44137043cd'`

Reply:
This is coming from the

```python
delayed_kwargs = dict(prefix=f"{dataset_id}.{table_id}-")
```

line earlier in this function. Ultimately we'll want to move away from `delayed` and constructing the Dask graph ourselves, so for now I think it's okay to drop the `prefix=` here and use

```python
delayed_kwargs = {}
```

instead. That should allow you to also drop the `compute()` call in `assert_eq`